pkgs/top-level/splice.nix

```diff
@@ -146,6 +146,21 @@ in
 
   newScope = extra: lib.callPackageWith (splicedPackagesWithXorg // extra);
 
+  callPackageWithExtraSet = extraSetString:
```

Comment:
it should take a `extraSetAttrpathList` but i had issues with converting the list from forEach to a set so i could `splicedPackages // extraSpliced` 

```
      extraSpliced = lib.forEach extraSetAttrpathList (extraSet: { name = extraSet; value = lib.attrByPath (lib.splitString "." extraSet) (throw "extraSet should be a attrpath relative to top-level") splicedPackages; });
```